### PR TITLE
TP-6435: Create generic log method in Dough

### PR DESCRIPTION
**Pull request by Guntrisoft**
which tests for console object before attempting to use it

### CODE REVIEW

File: assets/js/lib/utilities.js [hunk 1]
```diff
@@ -34,8 +34,26 @@ define('utilities', [], function() {
...
+        window.console[type ? type : 'log'](message);
+      }
+    },
+
+    /**
+     * Check to see if the console object exists
+     * @return {Boolean}
+     */
+    doesConsoleExist: function() {
+      return typeof console !== 'undefined';
```
Comment (benbarnett) [inline assets/js/lib/utilities.js]: Might need to also check if the `type` exists on `console`. E.g. I think some browsers have console.log but not console.warn. There's a bunch of others too.

File: spec/js/tests/utilities_spec.js [hunk 2]
```diff
@@ -41,4 +41,53 @@ describe('utilities', function() {
...
 
+  describe('log', function() {
+    var sandbox;
+
+    beforeEach(function() {
+      sandbox = sinon.sandbox.create();
+
+      // stub some console methods
+      sandbox.stub(window.console, "log");
+      sandbox.stub(window.console, "warn");
```
Comment (neoeno) [inline spec/js/tests/utilities_spec.js]: Not sure if you have a quotes standard here — but these are double while you've used single everywhere else

**Comment by neoeno** [comment]
Looks good :+1:

Where might us humble dough users be expected to use this? Anywhere we like? Or is it just for Dough internals?

**Comment by benbarnett** [comment]
@neoeno anywhere you like – just need to make sure `utilities` has been required in.

File: spec/js/tests/utilities_spec.js (hunk 2 above)
Comment (Guntrisoft) [inline spec/js/tests/utilities_spec.js]: fair point! - will alter!

**Comment by benbarnett** [comment]
:+1:

**Pull request merged**